explorer/dockerExplorer.ts

```diff
@@ -0,0 +1,163 @@
+    }
+    
+    refreshContainers(): void {
+        this._onDidChangeTreeData.fire(this._containersNode);
+    }
+
+    refreshRegistries(): void {
+        //     this._onDidChangeTreeData.fire(this._registriesNode);
+    }
+
+    private setAutoRefresh(): void {
+
+        const configOptions: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration('docker');
+        const interval = configOptions.get('explorerRefreshInterval', 1000);
+
+        if (interval > 0) {
+            clearTimeout(this._debounceTimer);
+            this._debounceTimer = setTimeout(() => {
+                this.refresh();
+            }, interval);
+        }
+    }
+
+    getTreeItem(element: DockerNode): vscode.TreeItem {
+        return element;
+    }
+
+    async getChildren(element?: DockerNode): Promise<DockerNode[]> {
+        return this.getDockerNodes(element);
+    }
+
+    private async getDockerNodes(element?: DockerNode): Promise<DockerNode[]> {
+
+        let opts = {};
+        let iconPath: any = {};
+        let contextValue: string = "";
+        let node: DockerNode;
+        const nodes: DockerNode[] = [];
+
+        if (!element) {
+            this._imagesNode = new DockerNode("Images", vscode.TreeItemCollapsibleState.Collapsed, "dockerImagesLabel", null, null);
+            this._containersNode = new DockerNode("Containers", vscode.TreeItemCollapsibleState.Collapsed, "dockerContainersLabel", null, null);
+            // this._registriesNode = new DockerNode("Registries", vscode.TreeItemCollapsibleState.Collapsed, "dockerLabelRegistries", null, null);
+            nodes.push(this._imagesNode);
+            nodes.push(this._containersNode);
+            // nodes.push(this._registriesNode);
+        } else {
+
+            if (element.label === 'Images') {
+                const images: Docker.ImageDesc[] = await docker.getImageDescriptors();
+                if (!images || images.length == 0) {
+                    return [];
+                } else {
+                    for (let i = 0; i < images.length; i++) {
+                        contextValue = "dockerImage";
+                        if (!images[i].RepoTags) {
+                            let node = new DockerNode("<none>:<none>", vscode.TreeItemCollapsibleState.None, contextValue);
+                            node.imageDesc = images[i];
+                            nodes.push(node);
+                        } else {
+                            for (let j = 0; j < images[i].RepoTags.length; j++) {
+                                let node = new DockerNode(images[i].RepoTags[j], vscode.TreeItemCollapsibleState.None, contextValue);
+                                node.imageDesc = images[i];
+                                nodes.push(node);
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (element.label === 'Containers') {
+
+                opts = {
```

Comment:
as noted above, I'm just leaving some options open for what we show in the tree.